BrowserKit/Sources/ComponentLibrary/Buttons/SecondaryRoundedButton.swift

```diff
@@ -5,54 +5,90 @@
 import Common
 import UIKit
 
-public class SecondaryRoundedButton: LegacyResizableButton, ThemeApplicable {
+public class SecondaryRoundedButton: ResizableButton, ThemeApplicable {
     private struct UX {
         static let buttonCornerRadius: CGFloat = 12
         static let buttonVerticalInset: CGFloat = 12
         static let buttonHorizontalInset: CGFloat = 16
         static let buttonFontSize: CGFloat = 16
-    }
-
-    private var highlightedTintColor: UIColor!
-    private var normalTintColor: UIColor!
 
-    override open var isHighlighted: Bool {
-        didSet {
-            backgroundColor = isHighlighted ? highlightedTintColor : normalTintColor
-        }
+        static let contentInsets = NSDirectionalEdgeInsets(
+            top: buttonVerticalInset,
+            leading: buttonHorizontalInset,
+            bottom: buttonVerticalInset,
+            trailing: buttonHorizontalInset
+        )
     }
 
+    private var highlightedBackgroundColor: UIColor!
+    private var normalBackgroundColor: UIColor!
+    private var foregroundColor: UIColor!
+
     override init(frame: CGRect) {
         super.init(frame: frame)
 
-        titleLabel?.font = DefaultDynamicFontHelper.preferredBoldFont(
-            withTextStyle: .callout,
-            size: UX.buttonFontSize)
-        layer.cornerRadius = UX.buttonCornerRadius
-        titleLabel?.textAlignment = .center
+        configuration = UIButton.Configuration.filled()
         titleLabel?.adjustsFontForContentSizeCategory = true
-        contentEdgeInsets = UIEdgeInsets(top: UX.buttonVerticalInset,
-                                         left: UX.buttonHorizontalInset,
-                                         bottom: UX.buttonVerticalInset,
-                                         right: UX.buttonHorizontalInset)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override public func updateConfiguration() {
+        guard var updatedConfiguration = configuration else {
+            return
+        }
+
+        updatedConfiguration.background.backgroundColor = switch state {
+        case [.highlighted]:
+            highlightedBackgroundColor
+        default:
+            normalBackgroundColor
+        }
+        updatedConfiguration.baseForegroundColor = foregroundColor
+
+        configuration = updatedConfiguration
     }
 
     public func configure(viewModel: SecondaryRoundedButtonViewModel) {
+        guard var updatedConfiguration = configuration else {
+            return
+        }
+
+        updatedConfiguration.titleTextAttributesTransformer = UIConfigurationTextAttributesTransformer { incoming in
```

Comment:
Unfortunately it's not working if you add the `titleTextAttributesTransformer` closure here. It works fine however if it's in `updateConfiguration`. Here is what it does in the current state of the PR:

https://github.com/mozilla-mobile/firefox-ios/assets/4530/cfb33461-f01d-4def-8683-fee33148d5ea